exercises/practice/custom-set/CustomSet.pm

```diff
@@ -1,55 +1,54 @@
 package CustomSet;
 
-use strict;
-use warnings;
+use Moo;
 use feature qw<say>;
 
-use Exporter qw<import>;
-our @EXPORT_OK
-    = qw<is_empty_set set_contains is_subset is_disjoint_set is_equal_set add_set_element set_intersection set_difference set_union>;
+has elements => (
+    is => 'ro',
+);
```

Comment:
The main thing I'd be worried about is whether a new developer doing this problem would know of `coerce`.

I would probably add a comment hinting towards the existence of it, á la

```perl
# The object is initialized with CustomSet->new(elements => [1,2,3,4]);
#
# If you wish to use a different underlying representation than an array
# reference, you can automatically convert the input by specifying a
# coercion (https://metacpan.org/pod/Moo#coerce) for the field:
#
# coerce => sub { my ($elements) = @_; ... },
```

Don't know if this is too verbose.

Alternatively something like

```perl
has elements => (
    is => 'ro',
    # Use coerce if a different representation is desired.
    # https://metacpan.org/pod/Moo#coerce
    # coerce => sub { my ($elements) = @_; ... },
);
```

Reply:
I've modified the exercise to use the student's own implementation of `is_equal_to` instead of direct object comparisons. May still be a good idea to encourage the use of something along the lines of `coerce`, but using `is_equal_to` removes the necessity of implementing this sort of modification.